// @sanity/base/lib/theme/color/input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.input = void 0;

var _sanityCssCustomProperties = _interopRequireDefault(require("sanity:css-custom-properties"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var input = {
  default: {
    enabled: {
      bg: _sanityCssCustomProperties.default['--input-bg'],
      fg: _sanityCssCustomProperties.default['--input-color'],
      border: _sanityCssCustomProperties.default['--input-border-color'],
      placeholder: _sanityCssCustomProperties.default['--input-color-placeholder']
    },
    disabled: {
      bg: _sanityCssCustomProperties.default['--input-bg-disabled'],
      fg: _sanityCssCustomProperties.default['--input-color-read-only'],
      border: _sanityCssCustomProperties.default['--input-border-color'],
      placeholder: _sanityCssCustomProperties.default['--input-color-placeholder']
    },
    hovered: {
      bg: _sanityCssCustomProperties.default['--input-bg'],
      fg: _sanityCssCustomProperties.default['--input-color'],
      border: _sanityCssCustomProperties.default['--input-border-color-hover'],
      placeholder: _sanityCssCustomProperties.default['--input-color-placeholder']
    }
  },
  invalid: {
    enabled: {
      bg: _sanityCssCustomProperties.default['--input-bg-invalid'],
      fg: _sanityCssCustomProperties.default['--input-color'],
      border: _sanityCssCustomProperties.default['--input-border-color-invalid'],
      placeholder: _sanityCssCustomProperties.default['--input-color-placeholder']
    },
    disabled: {
      bg: _sanityCssCustomProperties.default['--input-bg-invalid'],
      fg: _sanityCssCustomProperties.default['--input-color'],
      border: _sanityCssCustomProperties.default['--input-border-color-invalid'],
      placeholder: _sanityCssCustomProperties.default['--input-color-placeholder']
    },
    hovered: {
      bg: _sanityCssCustomProperties.default['--input-bg-invalid'],
      fg: _sanityCssCustomProperties.default['--input-color'],
      border: _sanityCssCustomProperties.default['--input-border-color-invalid'],
      placeholder: _sanityCssCustomProperties.default['--input-color-placeholder']
    }
  }
};
exports.input = input;
